### ISSUE TITLE
["Request"] min and max methods for NonEmptyList

### ISSUE THREAD

**Issue by dorians** [arrow-kt/arrow#2593]
### What version are you currently using?
1.0.1

### What would you like to see?
Kotlin's `Iterable` has methods `minOrNull` and `maxOrNull`:

```
listOf(1, 2, 3, 4, 5).minOrNull() // 1
listOf(1, 2, 3, 4, 5).maxOrNull() // 5
```

As a input list can be empty, they return nullable type:
```
emptyList<Int>().minOrNull() // null
```

I would be useful to have similar methods for `NonEmptyList` that return non-nullable values:
```
fun <T> NonEmptyList<T>.min(): T
fun <T> NonEmptyList<T>.max(): T
```

Other related methods:
* minByOrNull
* maxByOrNull

**Comment by nomisRev** [comment]
Hey @dorians,

That'd be great! I think that method should be restricted to `Comparable<T>`, no?

```kotlin
fun <T : Comparable<T>> NonEmptyList<T>.min(): T
fun <T : Comparable<T>> NonEmptyList<T>.max(): T
```

If you're interested we'd love the contribution!

**Comment by dorians** [comment]
@nomisRev yes, you're correct! I'll create PR for this one

**Comment by serras** [comment]
Closed by #2622
